src/main/java/help/swgoh/api/SwgohAPI.java

```diff
@@ -111,360 +105,574 @@ public static Optional<Language> fromUnicode( String unicode )
     }
 
     /**
      * This registry is a single-sourced discord-to-ally code registration for all patreon-tier users.
-     *
+     * <p>
      * Fetch registrations to the pool for for the specified ally code(s).
-     *
+     * <p>
      * https://api.swgoh.help/patreon
      *
      * @param allyCodes List of ally codes to search
      * @return registered pairing of ally code / discord ID, if any.
      */
-    CompletableFuture<RegistrationResponse> getRegistrationByAllyCodes( List<Integer> allyCodes );
-    default CompletableFuture<RegistrationResponse> getRegistrationByAllyCode( int allyCode )
-    {
-        return getRegistrationByAllyCodes( Collections.singletonList( allyCode ) );
+    CompletableFuture<RegistrationResponse> getRegistrationByAllyCodes(List<Integer> allyCodes);
+
+    default CompletableFuture<RegistrationResponse> getRegistrationByAllyCode(int allyCode) {
+        return getRegistrationByAllyCodes(Collections.singletonList(allyCode));
     }
 
     /**
      * This registry is a single-sourced discord-to-ally code registration for all patreon-tier users.
-     *
+     * <p>
      * Fetch registrations to the pool for for the specified discord ID(s).
-     *
+     * <p>
      * The discord "snowflake" ID format is specified here: https://discordapp.com/developers/docs/reference#snowflakes
-     *
+     * <p>
      * https://api.swgoh.help/patreon
      *
      * @param discordIds List of discord IDs to search
      * @return registered pairing of ally code / discord ID, if any.
      */
-    CompletableFuture<RegistrationResponse> getRegistrationByDiscordIds( List<String> discordIds );
-    default CompletableFuture<RegistrationResponse> getRegistrationByDiscordId( String discordId )
-    {
-        return getRegistrationByDiscordIds( Collections.singletonList( discordId ) );
+    CompletableFuture<RegistrationResponse> getRegistrationByDiscordIds(List<String> discordIds);
+
+    default CompletableFuture<RegistrationResponse> getRegistrationByDiscordId(String discordId) {
+        return getRegistrationByDiscordIds(Collections.singletonList(discordId));
+    }
+
+
+    /**
+     * Returns a list of player objects which are identified by the given ally code(s).
+     * <p>
+     * Cache sync:
+     * Registered user: 4 hours
+     * Verified user: 2 hours
+     * Patreon user: 1 hour
+     * <p>
+     * https://api.swgoh.help/swgoh
+     *
+     * @param allyCodes Ally code(s) of the players to request.
+     * @param language  Optional language to return translated names. If no language specified, no translations will be applied.
+     * @param filter    Optional projection of response fields you want returned. If no fields are specified, all fields will be returned.
+     * @return player data by individual or group of ally codes.
+     */
+    CompletableFuture<List<Player>> getFullPlayers(List<Integer> allyCodes, Language language, SwgohAPIFilter filter);
```

Comment:
I'm debating whether the "getFull-" prefix is the best choice of terminology for these new methods. After all, the methods that return raw JSON will always have the full amount of data as well.

Perhaps a "-Object" or "-Objects" suffix would be better? Like `getPlayerObjects()`. Something that would indicate a typed object response rather than the String JSON response from the other methods.

What do you think?

Reply:
Wasn't happy with the name either but didn't find a good solution but "getWhateverObject(s)" seems like a good idea. Will change that today or tomorrow as soon as I got some free time.